packages/batch-tx-handler/src/l1-batch-handler/utils.ts

```diff
@@ -107,21 +147,21 @@ export const decodeL2Msgs = (l2Msgs: Uint8Array): string[] => {
       txHash.push(...decodeL2Msgs(nextData));
       current = endOfNext;
     }
+  } else if (kind === delayedMsgToBeAdded) {
+    const remainData: Uint8Array = l2Msgs.subarray(1);
+    const currentHash = ethers.utils.hexlify(remainData);
+    txHash.push(currentHash);
   }
   return txHash;
 };
 
 // Get related sequencer batch data from a sequencer batch submission transaction.
-export const getRawData = async (
-  sequencerTx: string,
-  l2NetworkId: number,
-  provider: ethers.providers.JsonRpcProvider,
-): Promise<Uint8Array> => {
+export const getRawData = async (sequencerTx: string): Promise<[Uint8Array, BigNumber]> => {
   //Because current arbitrum-sdk doesn't support latest sequencer inbox contract, so we use ethersjs here directly.
```

Comment:
Have you tried with the latest SDK version? (I noticed that you have 3.1.2 in yarn.lock)

Reply:
Should we merge this first and have another pr to update sdk version?